modify eachdist.py so `develop` subcommand works by default

pmcollins opened the pull request:
# Description

By default and with a clean checkout,`eachdist.py develop` fails with a `no matching distribution` error.

However, if you modify the script, setting [with_test_deps](https://github.com/open-telemetry/opentelemetry-python/blob/191b21a49d21c78eb4ad561835aef9429605ad82/scripts/eachdist.py#L211) to False, the script issues a pip command that does not indicate `[test]` extras for each package, and it completes successfully.

This PR makes a small change to `eachdist.py`, defaulting `with_test_deps` to `False` and adding a flag so users can still turn this capability on.

Possible workaround for #1846

Related to #1462

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Tested manually by running `eachdist.py develop` in a clean Python environment.

# Does This PR Require a Contrib Repo Change?

Answer the following question based on these examples of changes that would require a Contrib Repo Change:
- [The OTel specification](https://github.com/open-telemetry/opentelemetry-specification) has changed which prompted this PR to update the method interfaces of `opentelemetry-api/` or `opentelemetry-sdk/`
- The method interfaces of `test/util` have changed
- Scripts in `scripts/` that were copied over to the Contrib repo have changed
- Configuration files that were copied over to the Contrib repo have changed (when consistency between repositories is applicable) such as in
    - `pyproject.toml`
    - `isort.cfg`
    - `.flake8`
- When a new `.github/CODEOWNER` is added
- Major changes to project information, such as in:
    - `README.md`
    - `CONTRIBUTING.md`

- [x] Yes?
- [ ] No.

# Checklist:

- [x] Followed the style guidelines of this project
- [ ] Changelogs have been updated
- [ ] Unit tests have been added
- [ ] Documentation has been updated

pmcollins commented:
> Scripts in scripts/ that were copied over to the Contrib repo have changed

If this change is accepted happy to apply it to contrib as well.

ocelotl commented:
@pmcollins curious, what is the motivation behind this change? Is there something this script can do that `tox` can't?

pmcollins commented:
> @pmcollins curious, what is the motivation behind this change? Is there something this script can do that `tox` can't?

Hi @ocelotl -- I use the eachdist script to install all OTel python sub-projects in editable mode (`python eachdist.py develop`), and it looks like eachdist was developed for this purpose (among others). Do you feel that this approach is deprecated (or should it be)? I'm curious about alternative approaches (happy to discuss in Slack if you want, too).

In any case, it looks like my original problem is no longer a problem 🤷, but there is a new problem and that is that at -- at least for me -- the import order defined in eachdist.ini needs tweaking. I have updated this PR with that tweak, which is a little bit sneaky because we're repurposing this PR a bit, but at least now we have the option.

In its current form, `python scripts/eachdist.py develop` fails, but it succeeds with the changes proposed in this PR (I have updated this PR with a different order in eachdist.ini), so if these changes are good, I'd like to make them available to other developers. On the other hand, if we want to deprecate the `develop` subcommand or the whole script, happy to discuss that too.
